cdap-proto/src/main/java/co/cask/cdap/proto/WorkflowNodeStateDetail.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.proto;
+
+import co.cask.cdap.api.workflow.NodeStatus;
+import co.cask.cdap.api.workflow.Workflow;
+
+import javax.annotation.Nullable;
+
+/**
+ * Class to represent the state of the node in the {@link Workflow}.
+ */
+public final class WorkflowNodeStateDetail {
+
+  private final String nodeId;
+  private final NodeStatus nodeStatus;
+  private final String runId;
+  private final DefaultThrowable failureCause;
+
+  /**
+   * Create a new instance.
+   * @param nodeId id of the node inside the Workflow
+   * @param nodeStatus status of the node
+   * @param runId run id assigned to the node, null if current node represents custom action or predicate
+   * @param failureCause cause of failure, null if execution of the node succeeded
+   */
+  public WorkflowNodeStateDetail(String nodeId, NodeStatus nodeStatus, @Nullable String runId,
+                                 @Nullable DefaultThrowable failureCause) {
```

Comment:
This need to be restricted to `DefaultThrowable` since it is used for storing in mds. Added another version of method which takes two `@Nullable` params.